packages/vscode-extension/src/project/applicationSession.ts

```diff
@@ -125,14 +142,44 @@ export class ApplicationSession implements Disposable {
     private readonly applicationContext: ApplicationContext,
     private readonly device: DeviceBase,
     private readonly metro: MetroLauncher,
-    private readonly devtools: Devtools,
+    private readonly devtoolsServer: DevtoolsServer,
     private readonly packageNameOrBundleId: string,
     private readonly supportedOrientations: DeviceRotation[]
```

Comment:
don't like the fact that both device and application sessions own and maintain its own devtools instance. Especially that device session own application session. What is the purpose of that? Would it make sense for these to "share" that instance (meaning they won't both listed to onconnection calls). Or perhaps we need some abstraction for "devtools session" that abstracts away individual connections but can support re-connecting upol reloads?